src/V3Coverage.cpp

```diff
@@ -290,7 +291,28 @@ class CoverageVisitor final : public VNVisitor {
     // we can cover expressions in while loops, but the counting goes outside
     // the while, see: "minimally-intelligent decision about ... clock domain"
     // in the Toggle Coverage docs
-    void visit(AstWhile* nodep) override { iterateProcedure(nodep, false); }
+    void visit(AstWhile* nodep) override {
+        VL_RESTORER(m_state);
+        VL_RESTORER(m_inToggleOff);
+        m_inToggleOff = true;
```

Comment:
Shouldn't this only be set on the condition - why no toggle analysis in the statements?